src/main/java/UserInterface.java

```diff
@@ -0,0 +1,101 @@
+                " \\__,_|_|   \\__\\___|_| |_| |_|_|___/" + System.lineSeparator();
+        System.out.println("======================================" + System.lineSeparator());
+        System.out.println(logo);
+    }
+
+    public static void printGoodbye() {
+        System.out.println("======================================");
+        System.out.println("goodbye! hope to see you again soon!");
+        System.out.println("======================================");
+        scan.close();
+    }
+
+    public static void requestUsername() {
+        System.out.println("hello! i'm artemis. what is your name?");
+        System.out.print("[unknown user]: ");
+        username = scan.nextLine();
+    }
+
+    public static void printList() {
+        if (TaskHandler.listCount == 0) {
+            System.out.println("[Artemis]: Your list is empty!");
+            return;
+        }
+
+        System.out.println("[Artemis]: Your list is as such:");
+        for (int index = 0; index < TaskHandler.listCount; index++) {
+            Task currentTask = TaskHandler.taskList[index];
+            System.out.printf("%d. %s\n", index+1, currentTask);
+        }
+    }
+
+    public static void printHelp() {
+        System.out.printf("welcome to your personal list, %s!%s", username, System.lineSeparator());
+        System.out.println("usage: todo [item]");
+        System.out.println("       deadline [item] /by [due date]");
+        System.out.println("       event [item] /from [start] /to [end]");
+        System.out.println("you may also enter \"list\" to see your current list.");
+        System.out.println("you may also exit by entering \"bye\".");
+
+    }
+
+    public static void commandLine() {
+        String userInput;
+        Task currentTask;
+
+        while (true) {
+            System.out.printf("[%s]: ", username);
+             userInput = scan.nextLine().trim();
+             Parser.Command commandGiven = Parser.parseCommand(userInput);
+
+             switch (commandGiven) {
+             case TODO:
+                 currentTask = TaskHandler.createToDo(userInput);
+                 break;
+             case DEADLINE:
+                 currentTask = TaskHandler.createDeadline(userInput);
+                 break;
+             case EVENT:
+                 currentTask = TaskHandler.createEvent(userInput);
+                 break;
+             case LIST:
+                 printList();
+                 continue;
+             case MARK:
+             case UNMARK:
+                 Object[] parsedContent = Parser.parseMarkUnmark(userInput);
+                 if (parsedContent[1] == null) {
+                     System.out.println("[artemis]: invalid command. please enter \"[mark/unmark] <list item number>\"");
+                 }
+
+                 int markItem = (int) parsedContent[0];
+                 boolean isMarked = (boolean) parsedContent[1];
```

Comment:
Nit: should this variable be named isMark instead, since the code is parsing whether the command is to mark/unmark a task instead of parsing whether the task is marked/unmarked?

Reply:
Hmm, I intended it to be a "is it going to be marked or unmarked", so that was the rationale behind the name